processor/src/main/java/org/mapstruct/ap/internal/processor/MethodRetrievalProcessor.java

```diff
@@ -485,7 +486,7 @@ private boolean checkParameterAndReturnType(ExecutableElement method, List<Param
 
         Type parameterType = sourceParameters.get( 0 ).getType();
 
-        if ( parameterType.isIterableOrStreamType() && !resultType.isIterableOrStreamType() ) {
+        if ( !resultType.isIterableOrStreamType() && isStreamTypeOrIterableFromJavaStdLib( parameterType ) ) {
```

Comment:
Let's keep the order of the checks same as before, first the parameter and then the result type. It makes it clearer when reading the message as well.